### PR TITLE
use throwsA matcher instead of try-catch-fail

### PR DESCRIPTION
**Pull request by a14n**
Follow up on #82238

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart [hunk 1]
```diff
@@ -210,24 +210,23 @@ void main() {
...
-            'tool-command-max-rss',
-            'dummy',
-            label: 'fail',
-            value: 10,
-          ),
-        ]);
-      }
+      await expectLater(
+        () => flutterCommand.run(),
+        throwsA(isA<ToolExit>()),
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:215]: `throwsToolExit`

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:215]: oh wait nvm carry on

File: packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart [hunk 2]
```diff
@@ -435,18 +434,18 @@ void main() {
...
-          const TestTimingEvent(
-            'flutter',
-            'dummy',
-            Duration(milliseconds: 1000),
-            label: 'fail',
-          )));
-      }
+      await expectLater(
+        () => flutterCommand.run(),
+        throwsA(isA<ToolExit>()),
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:439]: `throwsToolExit`

**Review by jonahwilliams — COMMENTED** [review]
Tool changes LGTM

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by a14n** [comment]
@jonahwilliams regarding your comment on `throws*` I planned to make another PR with that changes (everywhere and not only on the changes of this PR).

**Pull request merged**